golang/cosmos/ante/inbound.go

```diff
@@ -67,24 +81,49 @@ func (ia inboundAnte) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next
 				return ctx, err
 			}
 		}
+		isPriority, err := ia.isPriorityMessage(ctx, msg)
+		if err != nil {
+			return ctx, err
+		}
+		if isPriority {
+			if hasNonPriority {
+				defer incStat()
+				return ctx, ErrInboundQueueFull
```

Comment:
So it looks like if a Tx has both priority and non-priority messages, it gets rejected, regardless of the fullness of the queues? Do we have a statement of the desired semantics anywhere?

Reply:
I made a call here given that the intent for prioritization is to restrict it to economic activity, and that senders should not mix concerns in the same transaction anyway. If we had a way to set context per message, I would lift that restriction, but that doesn't seem possible in the current cosmos-sdk